src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/yml/http_client_mock_response_factory.yml

```diff
@@ -6,4 +6,4 @@ framework:
         log: true
     http_client:
         default_options: ~
-        mock_response_factory: my_response_factory
+        mock_response_factory: true
```

Comment:
I don't understand properly the `true`. Do you have a pending documentation PR that could help me understand ?

Reply:
Yes I think a dedicated parameter is better because I do not find intuitive to set this to true to delegate at runtime during test to be able to provide the factory in question. Maybe some other approach would be a better fit ? Don't have an idea at the moment though.

Reply:
@Neirda24 maybe the problem is the parameter name in general?

If we renamed the parameter name to `mock_responses` then it could work the same way going forward. The problem then is backward compatibility which is harder to solve.

IMHO The right way would be be to change the parameter name, BUT due to backwards compability (as I understand Symfony's policy) this would require some extra gluecode to keep both options. One of the reasons I'm inclined to keep this current PRs solution is that it requires less code and tests than two options. 

I'm more than willing to do a documentation PR for it though :)